Listeners:
- address:
    socketAddress:
      address: '::'
      ipv4Compat: true
      portValue: 8080
  filterChains:
  - filters:
    - name: envoy.filters.network.http_connection_manager
      typedConfig:
        '@type': type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
        httpFilters:
        - name: envoy.filters.http.router
          typedConfig:
            '@type': type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
        mergeSlashes: true
        normalizePath: true
        rds:
          configSource:
            ads: {}
            resourceApiVersion: V3
          routeConfigName: http
        statPrefix: http
        use_remote_address: true
    name: http
  name: http
Routes:
- ignorePortInHostMatching: true
  name: http
  virtualHosts:
  - domains:
    - bar.example.com
    name: http~bar_example_com
    routes:
    - match:
        headers:
        - exactMatch: canary
          name: env
        prefix: /
      name: http~bar_example_com-route-0-httproute-bar-route-default-0-0-matcher-0
      route:
        cluster: kube_default_bar-svc-canary_80
        clusterNotFoundResponseCode: INTERNAL_SERVER_ERROR
    - match:
        prefix: /
      name: http~bar_example_com-route-1-httproute-bar-route-default-1-0-matcher-0
      route:
        cluster: kube_default_bar-svc_80
        clusterNotFoundResponseCode: INTERNAL_SERVER_ERROR
  - domains:
    - example.com
    name: http~example_com
    routes:
    - match:
        prefix: /
      name: http~example_com-route-0-httproute-example-route-default-0-0-matcher-0
      route:
        cluster: kube_default_example-svc_80
        clusterNotFoundResponseCode: INTERNAL_SERVER_ERROR
  - domains:
    - foo.example.com
    name: http~foo_example_com
    routes:
    - match:
        pathSeparatedPrefix: /login
      name: http~foo_example_com-route-0-httproute-foo-route-default-0-0-matcher-0
      route:
        cluster: kube_default_foo-svc_80
        clusterNotFoundResponseCode: INTERNAL_SERVER_ERROR
